commons/src/main/java/io/aiven/kafka/tieredstorage/commons/storage/filesystem/FileSystemStorage.java

```diff
@@ -64,19 +72,28 @@ public InputStream fetch(final String key, final int from, final int to) throws
             throw new IllegalArgumentException("from cannot be more than to, from=" + from + ", to=" + to + " given");
         }
 
-        final Path path = Path.of(fsRoot, key);
+        final Path path = fsRoot.resolve(key);
         final long fileSize = Files.size(path);
         if (to > fileSize) {
```

Comment:
Seems we have a bug here, since we should still throw an exception if `to == fileSize` and we need a test for this. Since you are adding all kinds of tests in this PR can you also fix this and add a test?